tbtc-ts/src/sweep.ts

```diff
@@ -0,0 +1,161 @@
+ *                      number of UTXOs and deposit data elements must equal.
+ * @param fee - the value that should be subtracted from the sum of the UTXOs
+ *              values and used as the transaction fee.
+ * @param walletPrivateKey - Bitcoin private key of the wallet.
+ * @param bitcoinClient - Bitcoin client used to interact with the network.
+ * @returns Empty promise.
+ */
+export async function sweepUtxos(
+  utxos: UnspentTransactionOutput[],
+  depositData: DepositData[],
+  fee: BigNumber,
+  walletPrivateKey: string,
+  bitcoinClient: BitcoinClient
+): Promise<void> {
+  if (utxos.length != depositData.length) {
+    throw new Error(
+      "Number of UTXOs must equal the number of deposit data elements"
+    )
+  }
+
+  const utxosWithRaw: (UnspentTransactionOutput & RawTransaction)[] = []
+  for (const utxo of utxos) {
+    const rawTransaction = await bitcoinClient.getRawTransaction(
+      utxo.transactionHash
+    )
+
+    utxosWithRaw.push({
+      ...utxo,
+      transactionHex: rawTransaction.transactionHex,
+    })
+  }
+
+  const transaction = await createSweepTransaction(
+    utxosWithRaw,
+    depositData,
+    fee,
+    walletPrivateKey
+  )
+
+  await bitcoinClient.broadcast(transaction)
+}
+
+/**
+ * Creates a Bitcoin P2WPKH sweep transaction.
+ * @param utxos - UTXOs that should be used as transaction inputs.
+ * @param depositData - data on deposits. Each elements corresponds to UTXO. The
+ *                      number of UTXOs and deposit data elements must equal.
+ * @param fee - the value that should be subtracted from the sum of the UTXOs
+ *              values and used as the transaction fee.
+ * @param walletPrivateKey - Bitcoin private key of the wallet.
+ * @returns Bitcoin sweep transaction in raw format.
+ */
+export async function createSweepTransaction(
+  utxos: (UnspentTransactionOutput & RawTransaction)[],
+  depositData: DepositData[],
+  fee: BigNumber,
+  walletPrivateKey: string
+): Promise<RawTransaction> {
+  const decodedWalletPrivateKey = wif.decode(walletPrivateKey)
+
+  const walletKeyRing = new bcoin.KeyRing({
+    witness: true,
+    privateKey: decodedWalletPrivateKey.privateKey,
+    compressed: decodedWalletPrivateKey.compressed,
+  })
+
+  const walletAddress = walletKeyRing.getAddress("string")
+
+  const inputCoins = []
+  let totalInputValue = BigNumber.from(0)
+
+  for (const utxo of utxos) {
+    inputCoins.push(
+      bcoin.Coin.fromTX(
+        bcoin.MTX.fromRaw(utxo.transactionHex, "hex"),
+        utxo.outputIndex,
+        -1
+      )
+    )
+    totalInputValue = totalInputValue.add(utxo.value)
+  }
+
+  const transaction = new bcoin.MTX()
+
+  transaction.addOutput({
+    script: bcoin.Script.fromAddress(walletAddress),
+    value: totalInputValue.toNumber(),
+  })
+
+  await transaction.fund(inputCoins, {
+    changeAddress: walletAddress,
+    hardFee: fee.toNumber(),
+    subtractFee: true,
+  })
+
+  for (let i = 0; i < transaction.inputs.length; i++) {
```

Comment:
If we decide to go with `resolveDepositScript` approach, this loop should only call this method for given input.

Reply:
Done in https://github.com/keep-network/tbtc-v2/pull/96/commits/eca941e65d82a3b6a79704ca55e10fafd72c8c46.